docs/modules/ROOT/pages/components.adoc

```diff
@@ -0,0 +1,186 @@
+
+=== Setting up
+
+:initializable-component: xref:/security.adoc#initializable[InitializableComponent]
+
+Integrating a component first requires that the contract declares the component with the `component!` macro:
+
+[,javascript]
+----
+#[starknet::contract]
+mod MyContract {
+    // Import the component
+    use openzeppelin::security::InitializableComponent;
+
+    // Declare the component
+    component!(path: InitializableComponent, storage: initializable, event: InitializableEvent);
+}
+----
+
+The `path` argument should be the imported component itself (in this case, {initializable-component}).
+Notice that the `storage` and `event` arguments are representations set within the macro.
+In other words, the `initializable` and `InitializableEvent` names follow the Contracts for Cairo convention, but they can be renamed.
+
+=== Accessing component state
+
+Before a contract can access a component's state, the contract must integrate the implementations that will be used.
+For example:
+
+[,javascript]
+----
+#[starknet::contract]
+mod MyContract {
+    use openzeppelin::security::InitializableComponent;
+
+    component!(path: InitializableComponent, storage: initializable, event: InitializableEvent);
+
+    // Gives the contract access to the implementation methods
+    impl InitializableImpl =
+        InitializableComponent::InitializableImpl<ContractState>;
+    impl InternalImpl = InitializableComponent::InternalImpl<ContractState>;
+}
+----
+
+Defining the ``impl``s give the contract access to the methods within those implementations from the component.
+For example, `initialize` is defined in the `InternalImpl`.
+It can, therefore, be called like this:
+
+[,javascript]
+----
+#[starknet::contract]
+mod MyContract {
+    (...)
+
+    fn initialize_contract(ref self: ContractState) {
+        self.initializable.initialize();
+    }
+}
+----
+
+TIP: Defining the `InternalImpl` in a contract also gives the contract indirect access to the component's `Storage`.
+
+=== Embedding implementations
+
+A contract can embed implementations into the ABI which will expose the methods of the implementation.
+To embed implementations, add the `#[abi(embed_v0)]` attribute above the `impl`:
+
+[,javascript]
+----
+#[starknet::contract]
+mod MyContract {
+    (...)
+
+    // This attribute exposes the methods of the `impl`
+    #[abi(embed_v0)]
+    impl InitializableImpl =
+        InitializableComponent::InitializableImpl<ContractState>;
+}
+----
+
+`InitializableImpl` defines the `is_initialized` method in the component.
+By adding the embed attribute, `is_initialized` becomes a contract entrypoint for `MyContract`.
+
+=== Component storage and events
+
+The component's storage and events must be added to the contract's `Storage` struct and `Event` enum respectively.
+If the component doesn't define any events, the compiler will still create an empty event enum inside the component module.
```

Comment:
i think this should come before the "accessing component state". this kindof belongs to Setting up